### DISCUSSION TITLE
JSON Parameters

### DISCUSSION THREAD

**Discussion by DavLa201** [powerbi-cli/powerbi-cli#40] [Q&A]
Hello all,
I am in the process of automating my Power BI workflow with PowerBI ClI. However, I have a problem. Json formats are often requested, but how do I know the parameters?

Example: pbicli report datasource update

What is the Name of the Update-Details parameter?

Thanks for the answers :)

**Comment by liprec** [comment]
Hi @DavLa201,

Thanks for the feedback and indeed the JSON format is missing.

It is a good start to add a reference to the Power BI API documentation for the JSON format definition. E.g., for `pbicli report datasource update` an example of the JSON input is: https://learn.microsoft.com/en-us/rest/api/power-bi/reports/update-datasources#examples

I ~~will create~~ *have created* an issue to add these references to the documentation: https://github.com/powerbi-cli/powerbi-cli.github.io/issues/4
